web/src/components/shared/CreateAd/CreateAdFormStep.tsx

```diff
@@ -0,0 +1,124 @@
+import { FC, ReactNode, useEffect, useRef, useState } from 'react';
+import { useTransition } from 'transition-hook';
+import cn from 'classnames';
+import { Box } from 'web/src/components/ui/Box';
+import { Text } from 'web/src/components/ui/Text';
+import * as s from './CreateAdFormStep.css';
+
+const ANIMATION_TIMEOUT = 400;
+
+interface Props {
+  title?: ReactNode;
+  children?: ReactNode;
+  activeValue?: ReactNode;
+  isActive?: boolean;
+  isCompleted?: boolean;
+  isCompletedBeforeActive?: boolean;
+  isNext?: boolean;
+  onClick: () => void;
+}
+
+export const CreateAdFormStep: FC<Props> = ({
+  title,
+  children,
+  activeValue,
+  isActive = false,
+  isCompleted = false,
+  isCompletedBeforeActive = false,
+  isNext = false,
+  onClick,
+}) => {
+  const contentRef = useRef<HTMLDivElement>(null);
+  const [height, setHeight] = useState<number | undefined>();
+  const { stage, shouldMount } = useTransition(isActive, ANIMATION_TIMEOUT);
+
+  useEffect(() => {
+    const contentContainer = contentRef.current;
+
+    function calcHeight() {
+      setHeight(contentRef.current?.scrollHeight);
+    }
```

Comment:
чтобы всё в одном стиле было
```suggestion
    const calcHeight = () => setHeight(contentRef.current?.scrollHeight);
```